src/core/array.d

```diff
@@ -0,0 +1,61 @@
+module core.array;
+
+/++
+    Returns a static array constructed from `a`. The type of elements can be
+    specified implicitly (`int[2] a = staticArray(1,2)`) or explicitly
+    (`float[2] a = staticArray!float(1,2)`).
++/
+pragma(inline, true) U[T.length] staticArray(U = CommonType!T, T...)(T a)
+{
+    return [a];
+}
+
+unittest
+{
+    {
+        auto a = staticArray(1, 2, 3);
+        assert(is(typeof(a) == int[3]));
+        assert(a == [1, 2, 3]);
+    }
+    {
+        auto a = staticArray(1, 2.0);
+        assert(is(typeof(a) == double[2]));
+        assert(a == [1, 2.0]);
+    }
```

Comment:
Simply create new unittest blocks instead of scoping. It doesn't hurt and allows the user of the documentation to hit "run" and "edit" for an individual block. Plus you can give a short summary title in `/// Amazing example`

Reply:
pushing back on this request as well, but happy to revisit if you have good arguments
* creates less visual clutter to group (and takes less vertical space both in rendered doc and in source code); the inline comments will still be visible in rendered docs
* (maybe this one isn't an important point) potentially less executable / compile time / runtime bloat